codegen/test/Spec.hs

```diff
@@ -136,8 +137,101 @@ nnSpec = do
     fs <- parseWith (Proxy @ NN')
     (length fs) `shouldBe` (length xs)
 
+  it "parses the `_thnn_binary_cross_entropy` function without inplace and buffer" $ do
+    fs <- parseWith (Proxy @ NN')
+
+    case fmap RNN.mkBackwardAndForward $ mhead $ filter (("_thnn_binary_cross_entropy" ==) . name . NN.func') fs of
+      Nothing -> fail "_thnn_binary_cross_entropy function not found!"
+      Just nf -> do
+        map (\v' -> Right (toTuple v')) nf `shouldBe`
+          map (\s -> fmap toTuple $ parseFuncSig s)
+          [ "_thnn_binary_cross_entropy_forward"
+            <> "(Tensor self, Tensor target, Tensor? weight={}, int64_t reduction=Reduction::Mean)"
+            <> " -> Tensor"
+          , "_thnn_binary_cross_entropy_forward_out"
+            <> "(Tensor output, Tensor self, Tensor target, Tensor? weight={}, int64_t reduction=Reduction::Mean)"
+            <> " -> Tensor"
+          , "_thnn_binary_cross_entropy_backward"
+            <> "(Tensor grad_output, Tensor self, Tensor target, Tensor? weight={}, int64_t reduction=Reduction::Mean)"
+            <> " -> Tensor"
+          , "_thnn_binary_cross_entropy_backward_out"
+            <> "(Tensor grad_input, Tensor grad_output, Tensor self, Tensor target, Tensor? weight={}, int64_t reduction=Reduction::Mean)"
+            <> " -> Tensor"
+          ]
+
+  it "parses the `_thnn_multilabel_margin_loss` function with buffer" $ do
+    fs <- parseWith (Proxy @ NN')
+
+    case fmap RNN.mkBackwardAndForward $ mhead $ filter (("_thnn_multilabel_margin_loss" ==) . name . NN.func') fs of
+      Nothing -> fail "_thnn_multilabel_margin_loss function not found!"
+      Just nf -> do
+        map (\v' -> Right (toTuple v')) nf `shouldBe`
+          map (\s -> fmap toTuple $ parseFuncSig s)
+          [ "_thnn_multilabel_margin_loss_forward"
+            <> "(Tensor self, LongTensor target, int64_t reduction=Reduction::Mean)"
+            <> " -> (Tensor, Tensor)"
+          , "_thnn_multilabel_margin_loss_forward_out"
+            <> "(Tensor output, Tensor is_target, Tensor self, LongTensor target, int64_t reduction=Reduction::Mean)"
+            <> " -> (Tensor, Tensor)"
+          , "_thnn_multilabel_margin_loss_backward"
+            <> "(Tensor grad_output, Tensor self, LongTensor target, int64_t reduction=Reduction::Mean, Tensor is_target)"
+            <> " -> Tensor"
+          , "_thnn_multilabel_margin_loss_backward_out"
+            <> "(Tensor grad_input, Tensor grad_output, Tensor self, LongTensor target, int64_t reduction=Reduction::Mean, Tensor is_target)"
+            <> " -> Tensor"
+          ]
+{-
+  it "parses the `_thnn_elu` function with inplace" $ do
```

Comment:
This function refers header's information.
Now, codegen does not support the function which needs header's one.